### PR TITLE
Some improvements to developer tools

### PR DESCRIPTION
**Pull request by Eightquake**
This wasn't related to any card in the Project, so no task id here or in the commits. I have made these changes while working on other cards in Project, but never committed them.

In this PR I have:
- Added installation of Redux and React developer tools for Electron windows using package [electron-devtools-installer](https://www.npmjs.com/package/electron-devtools-installer)
- Made a debug configuration for the Electron main process in VSCode
- Configured a background task for the react-scripts start process
- Updated README about these changes

I think it would be useful to have this in the repo. But, feel free to reject or comment your thoughts if you don't agree! There is no rush to merge this PR.

With these changes, a developer using VSCode could start the application by hitting F5. Breakpoints and other inspection works for the main process using this debug configuration. Also, restarting Electron is faster as Electron and React/webpack is independent of each other!
When the developer is in the Chrome DevTools for an electron window they could use the Redux devtools to look at the store or actions and React devtools to see the tree and all of the states.

I haven't tested this on a Windows machine, but I hope everything will work there.

***NOTE:*** After installing the Chrome extensions they will stay installed until you delete them from the Electron appdata folder!

### CODE REVIEW

**Comment by gustavjorlov** [comment]
Cool! I’ve thought about if this is possible but never investigated enough to know ;)

**Comment by Doverstav** [comment]
Redux and React devtools seem to install correctly, very useful!

Starting LogLady with F5 works, sort of... Runnnig it with F5 greets you with the following error message:
image

As can be seen in the image, npm seems to start succesfully, but the "frontend" process does not. However, hitting F5 one more time starts LogLady without problem. Is this intended behavior is there some issue here? Or am I using it incorrectly?

**Comment by Eightquake** [comment]
You are using it correctly, but I didn't get that error message on my end. The error message is from the launch/debug configuration, which is set to wait for the task to report begin - but it didn't before the timeout.

Earlier, I got that error message when the regex for the problem matcher wasn't correct - meaning the task couldn't know when it was ready. I fixed that, but maybe it works in some other way in Windows 🤔

Try editing the `.vscode/tasks.json` like this:
```diff
[...]
"problemMatcher": [
        {
          "pattern": [
            {
              [...]
            }
          ],
          "background": {
            "activeOnStart": false,
-            "beginsPattern": "^Compiled successfully!$",
+            "beginsPattern": "Compiled successfully!",
            "endsPattern": "."
          }
        }
      ]

**Comment by Doverstav** [comment]
Tried it, but I'm still getting the same error 😕

**Comment by Eightquake** [comment]
Alright, seems like I'll have to boot into Windows then! VSCode allows setting overrides for Windows specifically, maybe I can use that.

**Comment by Eightquake** [comment]
I've tested it now, and the startup of react-scripts start seems to be a bit slower on Windows, so it would timeout before webpack had compiled and was ready.
I found that after closing the background task and pressing F5 again it would work with the first regex. But re-opening VSCode and pressing F5 it would sometimes show the error message. So it is pretty inconsistent.

I tried updating the regex to match when it was starting as well. This seems to work all the time, with Electron always opening up with LogLady instead of an empty web page.

Could you try it on your computer and see if it works without a hiccup?
```diff
[...]
"problemMatcher": [
        {
          "pattern": [
            {
              [...]
            }
          ],
          "background": {
            "activeOnStart": false,
-            "beginsPattern": "^Compiled successfully!$",
+            "beginsPattern": "^(Compiled successfully!)|(Starting the development server...)$",
            "endsPattern": "."
          }
        }
      ]

**Comment by Doverstav** [comment]
The new regex seems to work for me as well! 👍

**Comment by gustavjorlov** [comment]
New regex sound there should be a new unit test as well 😎

**Comment by Eightquake** [comment]
Yeah, it is very heavily dependent on that Webpack or CRA doesn't change the messages. But how would one create tests for this kind of thing? 🤔

File: public/electron.js [hunk 1]
```diff
@@ -56,6 +61,23 @@ const windowStateKeeper = windowName => {
 };
 
 const createWindow = async () => {
+  // Install chrome extensions for React and Redux for the Electron windows
```
Comment (Doverstav) [inline public/electron.js]: Are these extensions installed on a "normal" users, i.e. when running the binaries? If they are, maybe that something we should avoid? With some `isDevMode()` check or something,

Comment (Eightquake) [reply public/electron.js]: Good catch! Yeah, they are installed even when not in dev-mode.
I'll update it :)

**Pull request merged**